fabric-renderer-api-v1/src/main/java/net/fabricmc/fabric/api/renderer/v1/material/RenderMaterial.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.renderer.v1.material;
+
+import net.fabricmc.fabric.api.renderer.v1.Renderer;
+import net.fabricmc.fabric.api.renderer.v1.mesh.MeshBuilder;
+import net.fabricmc.fabric.api.renderer.v1.mesh.MutableQuadView;
+import net.minecraft.block.Block;
+import net.minecraft.util.Identifier;
+
+/**
+ * All model quads have an associated render material governing
+ * how the quad will be rendered.<p>
+ * 
+ * A material instance is always immutable and thread-safe.  References to a material 
+ * remain valid until the end of the current game session.<p>
+ * 
+ * Materials can be registered and shared between mods using {@link Renderer#registerMaterial(net.minecraft.util.Identifier, RenderMaterial)}.
+ * The registering mod is responsible for creating each registered material at startup.<p>
+ * 
+ * Materials are not required to know their registration identity, and two materials
+ * with the same attributes may or may not satisfy equality and identity tests. Model 
+ * implementations should never attempt to analyze materials or implement control logic based on them.
+ * They are only tokens for communicating quad attributes to the ModelRenderer.<p>
+ * 
+ * There are three classes of materials... <p>
+ * 
+ * <b>STANDARD MATERIALS</b><p>
+ * 
+ * Standard materials have "normal" rendering with control over lighting,
+ * color, and texture blending. In the default renderer, "normal" rendering
+ * emulates unmodified Minecraft. Other renderers may offer a different aesthetic.<p>
+ * 
+ * The number of standard materials is finite, but not necessarily small. 
+ * To find a standard material, use {@link Renderer#materialFinder()}.<p>
+ * 
+ * All renderer implementations should support standard materials.<p>
+ * 
+ * <b>SHADER MATERIALS</b><p>
```

Comment:
The docs for shader and special materials reference features that didn't make the cut and should probably be removed.